File: LeetCode/June Challenge/03-Two City Scheduling/solution.cpp
class Solution {
public:
    static bool comparator(vector<int>& a, vector<int>& b)
    {
        return (a[1]-a[0])>(b[1]-b[0]);
    }
    int twoCitySchedCost(vector<vector<int>>& costs) {
        int n=costs.size();
        sort(costs.begin(),costs.end(),comparator);
        
        int sum=0;
        for(int i=0;i<n;i++)
        {
            sum+= i>=n/2 ? costs[i][1] : costs[i][0];
        }
        return sum;
    }
};

File: LeetCode/June Challenge/03-Two City Scheduling/solution.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        ans, n = 0, len(costs) // 2
        diff = [(a - b) for a, b in costs]
        order_diff = sorted((value, i) for (i, value) in enumerate(diff))
        for value, i in order_diff[:n]:
            ans += costs[i][0]
        for value, i in order_diff[n:]:
            ans += costs[i][1]
        
        return ans
